Sources/XcodeProj/Objects/Files/PBXGroup.swift

```diff
@@ -187,17 +202,11 @@ public extension PBXGroup {
         }
         let groupPath = try fullPath(sourceRoot: sourceRoot)
 
-        if override, let existingFileReference = try projectObjects.fileReferences.first(where: {
-            // Optimization: compare lastComponent before fullPath compare
-            guard let fileRefPath = $0.value.path else {
-                return try filePath == $0.value.fullPath(sourceRoot: sourceRoot)
-            }
-            let fileRefLastPathComponent = fileRefPath.split(separator: "/").last!
-            if filePath.lastComponent == fileRefLastPathComponent {
-                return try filePath == $0.value.fullPath(sourceRoot: sourceRoot)
-            }
-            return false
-        }) {
+        if override, let existingFileReference = try self.existingFileReference(
+            from: projectObjects,
+            with: filePath,
+            sourceRoot: sourceRoot
```

Comment:
Having spent some time going round and round on this, trying to create comprehensive tests, whilst still being slightly confused by how `PBXSourceTree`s actually work, I finally realised that the functionality needed is actually already *here*. It's only necessary to extract is and create a public driver method @kwridan @damirdavletov.

Therefore I decided not to assume too much about my knowledge of this lib, or PBXProj in general, and simply assume that this functionality works and expose it as we require. Seems much simpler. I can't guarantee it previously worked, but it should not break anything either...?